backend/app/modules/multisig_signatures/service/offchain.py

```diff
@@ -29,6 +33,30 @@ def get_last_pending_signature(
             hash=signature_db.hash,
         )
 
+    def approve_pending_signatures(self, _: Context) -> list[Signature]:
+        pending_signatures = database.multisig_signature.get_all_pending_signatures()
+        approved_signatures = []
+
+        for pending_signature in pending_signatures:
+            confirmations = get_message_details(
+                pending_signature.hash, is_mainnet=self.is_mainnet
+            )["confirmations"]
+            threshold = int(
+                get_user_details(pending_signature.address, is_mainnet=self.is_mainnet)[
+                    "threshold"
+                ]
+            )
+
+            if len(confirmations) >= threshold:
+                pending_signature.status = SigStatus.APPROVED
+                db.session.commit()
```

Comment:
Shouldn't we place `commit` at the end of the function? What if it fails in this `for loop`, some of the confirmations will be approved and some other not?

Reply:
yup, it's better to do it atomically, changing 👌